src/host/cmdline.cpp

```diff
@@ -772,13 +769,12 @@ til::point CommandLine::_moveCursorRightByWord(COOKED_READ_DATA& cookedReadData)
         if (*NextWord == L' ')
         {
             // If the current character is space, skip to the next non-space character.
-            while (NextWord < BufLast)
+            while (++NextWord < BufLast)

```

Comment:
Interesting. This change will make us skip the first part of `NextWord` on entry to the loop; however, we did already check `*NextWord`. 

I thought that it could cause a read beyond the end of `BufLast`, where we were supposed to stop. It doesn't seem to (since the preincrement will result in dropping out of the loop once `BufLast` is reached)...

I had to stop to think about this, which in and of itself is not a bad thing ... but I do wonder if this version harms readability for most of our contributors. Code doesn't have to be clever to be effective :smile:

Reply:
Then again, it makes this block of code look like the block on line 785. That seems OK.